webapp/src/containers/modals/SubscribeModal/filters/repos.tsx

```diff
@@ -10,58 +10,89 @@ import usePluginApi from 'hooks/usePluginApi';
 import {formLabelValuePairs} from 'utils';
 
 type ReposFilterProps = {
-    handleSelectRepo: (repo: string, repoName?: string) => void
+    organization: string
+    projectId: string
+    eventType: string
     selectedRepo: string
+    handleSelectRepo: (value: string, name?: string) => void
     selectedTargetBranch: string
-    handleSelectTargetBranch: (branch: string) => void
-} & ReposSubscriptionFiltersRequest
+    handleSelectTargetBranch: (value: string, name?: string) => void
+    selectedPullRequestCreatedBy: string
+    handleSelectPullRequestCreatedBy: (value: string, name?: string) => void
+    selectedPullRequestReviewersContains: string
+    handlePullRequestReviewersContains: (value: string, name?: string) => void
+    selectedPushedBy: string
+    handleSelectPushedBy: (value: string, name?: string) => void
+    selectedMergeResult: string
+    handleSelectMergeResult: (value: string, name?: string) => void
+    selectedNotificationType: string
+    handleSelectNotificationType: (value: string, name?: string) => void
```

Comment:
You can optimize this by passing two props only - 
Something like this - 
```
selectedFilter : {filterName: string, filterValue: string},
handleSelectFilter: (filterName: string, value: string, name?; string) => void
```

Reply:
yes we can do something similar but here we will need to handle a few more things like in `handleSelectFilter: (filterName: string, value: string, name?; string) => void` we will need to target `name` field to put its value into the needed field only.

Have added a TODO for it, we can refactor it later after this release